stdlib/public/core/String.swift

```diff
@@ -437,44 +437,44 @@ extension String {
-  /// replacement character (`"\u{FFFD}"`); This may require resizing
+  /// replacement character (`"\u{FFFD}"`). This may require resizing
   /// the buffer beyond its original capacity.
   ///
   /// The following examples use this initializer with the contents of two
-  /// different `UInt8` arrays---the first with well-formed UTF-8 code unit
-  /// sequences and the second with an ill-formed sequence at the end.
+  /// different `UInt8` arrays---the first with a well-formed UTF-8 code unit
+  /// sequence, and the second with an ill-formed sequence at the end.
   ///
-  ///     let validUTF8: [UInt8] = [67, 97, 102, -61, -87, 0]
-  ///     let s = String(unsafeUninitializedCapacity: validUTF8.count,
-  ///                    initializingUTF8With: { ptr in
-  ///         ptr.initializeFrom(validUTF8)
+  ///     let   validUTF8: [UInt8] = [0x43, 0x61, 0x66, 0xC3, 0xA9]
+  ///     let invalidUTF8: [UInt8] = [0x43, 0x61, 0x66, 0xC3]
+  ///
+  ///     String(unsafeUninitializedCapacity: validUTF8.count,
+  ///            initializingUTF8With: { ptr in
+  ///         _ = ptr.initialize(from: validUTF8)
   ///         return validUTF8.count
   ///     })
-  ///     // Prints "Café"
+  ///     // $R0: String = "Café"
   ///
-  ///     let invalidUTF8: [UInt8] = [67, 97, 102, -61, 0]
-  ///     let s = String(unsafeUninitializedCapacity: invalidUTF8.count,
-  ///                    initializingUTF8With: { ptr in
-  ///         ptr.initializeFrom(invalidUTF8)
+  ///     String(unsafeUninitializedCapacity: invalidUTF8.count,
+  ///            initializingUTF8With: { ptr in
+  ///         _ = ptr.initialize(from: invalidUTF8)
   ///         return invalidUTF8.count
   ///     })
-  ///     // Prints "Caf�"
+  ///     // $R1: String = "Caf�"
   ///
-  ///     let s = String(unsafeUninitializedCapacity: invalidUTF8.count,
-  ///                    initializingUTF8With: { ptr in
-  ///         ptr.initializeFrom(invalidUTF8)
+  ///     String(unsafeUninitializedCapacity: invalidUTF8.count,
+  ///            initializingUTF8With: { ptr in
+  ///         _ = ptr.initialize(from: invalidUTF8)
   ///         return 0
   ///     })
-  ///     // Prints ""
+  ///     // $R2: String = ""
   ///
   /// - Parameters:
   ///   - capacity: The number of UTF-8 code units worth of memory to allocate
-  ///       for the String.
-  ///   - initializer: A closure that initializes elements and sets the count of
-  ///       the new String
-  ///     - Parameters:
-  ///       - buffer: A buffer covering uninitialized memory with room for the
-  ///           specified number of UTF-8 code units.
+  ///     for the String (excluding the null terminator).
+  ///   - initializer: A closure that initializes a given buffer, and returns
+  ///     the number of initialized elements (or 0 if it fails to initialize).
+  ///   - buffer: A buffer covering uninitialized memory with room for the
+  ///     specified number of UTF-8 code units (excluding the null terminator).
```

Comment:
The `buffer` parameter is the parameter to the `initializer` closure, not a parameter of the init itself, so it shouldn't appear here (documenting closure parameters doesn't work). How about this rewrite:

```suggestion
  ///   - initializer: A closure that accepts a buffer covering uninitialized
  ///     memory with room for `capacity` UTF-8 code units, initializes
  ///     that memory, and returns the number of initialized elements.
```